src/server/server.rs

```diff
@@ -451,13 +452,25 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
     }
 
     fn on_snap_token(&mut self, event_loop: &mut EventLoop<Self>, _: u64, token: Token) {
-        // Now we close every snapshot connection after sending successfully.
+        // We use new connection for every snapshot, and will close it after sending OK.
+        // But we can't close it here directly, because we can't guarantee that
+        // the socket data is sent to remote successfully. So we wait some time and
+        // then close it.
+        // TODO: we should use a better way to do this. 
+        if let Err(e) = event_loop.timeout_ms(Tick::GcConn { token: token },
+                                              SNAPSHOT_CONN_TIMEOUT) {
+            error!("register gc token {:?} err {:?}, remove directly", token, e);
+            self.remove_conn(event_loop, token);
```

Comment:
I don't think that we can use `self.remove_conn` here. Because it will remove the normal connection token too.